Kernel/System/SupportDataCollector/Plugin/OTRS/TimeSettings.pm

```diff
@@ -78,7 +78,14 @@ sub Run {
     }
 
     # Calendar time zones
-    for my $Counter ( 1 .. 9 ) {
+    my $Maximum = $ConfigObject->Get("MaximumCalendarNumber") || 50; 
+
+    COUNTER:
+    for my $Counter ( '', 1 .. $Maximum ) {
+        my $CalendarName = $ConfigObject->Get('TimeZone::Calendar' . $Counter . 'Name' );
+
+        next COUNTER if !$CalendarName;
+
         my $CalendarTimeZone = $ConfigObject->Get( 'TimeZone::Calendar' . $Counter );
```

Comment:
Hi @reneeb,

I have a small question...

Why do you check especially for the name, if only the CalendarTimeZone is interesting for the rest of the code?
Do you just want to make sure that a TimeZone and a name are configured for each calendar?

good job 🚀

Reply:
Probably the CalenderName is required to select a calender

Reply:
Without that check, the list of "missing" timezone settings would contain even those calendars that do not exist in the system configuration: Currently Znuny has 9 calendars, but the check would list 50 calendars in the check.

$ConfigObject->Get('....') doesn't croak if a setting doesn't exist at all, so we need another way to ensure to list only existing calendars.